src/winrtble/peripheral.rs

```diff
@@ -81,13 +86,66 @@ impl Peripheral {
             properties.manufacturer_data = manufacturer_data
                 .into_iter()
                 .map(|d| {
-                    let company_id = d.company_id().unwrap();
+                    let manufacturer_id = d.company_id().unwrap();
                     let buffer = d.data().unwrap();
                     let reader = DataReader::from_buffer(&buffer).unwrap();
                     let len = reader.unconsumed_buffer_length().unwrap() as usize;
                     let mut data = vec![0u8; len];
                     reader.read_bytes(&mut data).unwrap();
-                    (company_id, data)
+
+                    // Emit event of newly received advertisement
+                    self.adapter.emit(CentralEvent::ManufacturerAdvertisement {
+                        address: self.address,
+                        manufacturer_id,
+                        data: data.clone(),
+                    });
+
+                    (manufacturer_id, data)
+                })
+                .collect();
+        }
+
+        if let Ok(data_sections) = advertisement.data_sections() {
+            properties.service_data = data_sections
+                .into_iter()
+                .filter_map(|d| {
+                    let buffer = d.data().unwrap();
+                    let reader = DataReader::from_buffer(&buffer).unwrap();
+                    let len = reader.unconsumed_buffer_length().unwrap() as usize;
+                    let mut data = vec![0u8; len];
+                    reader.read_bytes(&mut data).unwrap();
```

Comment:
It might be worth factoring out a helper function to convert a buffer to a `Vec<u8>`, as there are at least two places doing the same thing.

Reply:
I've since created `utils::to_vec()`